assembl/models/notification.py

```diff
@@ -897,33 +898,58 @@ def render_to_email_text_part(self):
     def get_notification_subject(self):
         """Typically for email"""
 
-    def get_unlocalized_jinja_env(self):
-        threadlocals = self.threadlocals
-        if getattr(threadlocals, 'jinja_env', None) is None:
-            threadlocals.jinja_env = Environment(
+    @classmethod
+    def make_unlocalized_jinja_env(cls):
+        return Environment(
                 loader=PackageLoader('assembl', 'templates'),
                 extensions=['jinja2.ext.i18n'])
-        return threadlocals.jinja_env
 
-    def get_jinja_env(self):
-        jinja_env = self.get_unlocalized_jinja_env()
-        self.setup_localizer(jinja_env)
+    @classmethod
+    def make_jinja_env(cls, user=None):
+        jinja_env = cls.make_unlocalized_jinja_env()
+        cls.setup_localizer(jinja_env, user)
         return jinja_env
 
-    def get_localizer(self):
-        locale = self.first_matching_subscription.user.get_preferred_locale()
+    def get_jinja_env(self):
+        threadlocals = self.threadlocals
+        if getattr(threadlocals, 'jinja_env', None) is None:
+            threadlocals.jinja_env = self.make_unlocalized_jinja_env()
+        self.setup_localizer(
+            threadlocals.jinja_env, self.first_matching_subscription.user)
+        return threadlocals.jinja_env
+
+    @classmethod
```

Comment:
Yes, I see. The notification system was intended to be subscribed to, and then it sends out emails. Okay, I see why you did it. Hmmm, tough one. We're bending assumptions a bit. Want to discuss this with @swergas next week? I think for this PR this change is okay. But this shows a rigidity in the notification system that I wasn't expecting. I say, TBD later.